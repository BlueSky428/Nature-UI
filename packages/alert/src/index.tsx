import * as React from 'react';
import clx from 'clsx';
import { FiInfo, FiAlertCircle, FiAlertTriangle, FiCheckCircle } from 'react-icons/fi';

const SUBTLE_TEXT = 'text-gray-800';

export const ALERT_STATUSES = {
  error: {
    bg: 'bg-red-200',
    iconColor: 'text-red-600 mr-3',
    icon: FiAlertCircle,
    variant: {
      solid: 'bg-red-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
  info: {
    bg: 'bg-blue-200',
    iconColor: 'text-blue-600 mr-3',
    icon: FiInfo,
    variant: {
      solid: 'bg-blue-600',
      subtle: SUBTLE_TEXT,
    },
  },
  success: {
    bg: 'bg-green-200',
    iconColor: 'text-green-600 mr-3',
    icon: FiCheckCircle,
    variant: {
      solid: 'bg-green-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
  warning: {
    bg: 'bg-orange-200',
    iconColor: 'text-orange-600 mr-3',
    icon: FiAlertTriangle,
    variant: {
      solid: 'bg-orange-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
};

interface AlertOptions {
  /**
   * The status of the alert
   */
  status?: keyof typeof ALERT_STATUSES;
  /**
   * The variant of the alert style to use
   */
  variant?: 'solid' | 'subtle';

  /**
   * Other tailwind utility classes or custom classnames you wish to include
   */
  className?: string;

  children?: React.ReactNode;
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component?: React.ElementType;
}

export type AlertProps = AlertOptions;

const BASE_STYLE = 'px-4 py-3 flex items-center';

const Alert: React.ForwardRefExoticComponent<AlertProps> = React.forwardRef(
  (props: AlertProps, ref: React.Ref<any>) => {
    const {
      className = '',
      status = 'success',
      children,
      variant = 'subtle',
      component: Component = 'div',
    } = props;

    // const Component = 'div';

    const VARIANT: string = ALERT_STATUSES[status].variant[variant];

    const componentClass = clx(BASE_STYLE, {
      [className]: className,
      [ALERT_STATUSES[status].bg]: status,
      [VARIANT]: variant,
    });

    const Icon = ALERT_STATUSES[status];
    const iconClasses = clx({
      [VARIANT]: variant,
      'mr-3': variant,
      [Icon.iconColor]: variant !== 'solid',
    });

    const IconComponent = Icon.icon;

    return (
      <Component className={componentClass} ref={ref}>
        {status && <IconComponent className={iconClasses} size={20} />}
        <span>{children ? children : 'This is an alert ðŸ™‚'}</span>
      </Component>
    );
  }
);

Alert.displayName = 'Alert';

export default Alert;
