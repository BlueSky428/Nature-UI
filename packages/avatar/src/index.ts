export * from './avatar';
export * from './avatarGroup';
