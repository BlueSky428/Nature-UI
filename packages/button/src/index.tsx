import * as React from 'react';
import clx from 'clsx';

interface ButtonProps {
  /**
   * The label of the button
   */
  'aria-label'?: string;
  /**
   * The text color of the button. Use a color key passed in theme.colors.
   */
  text?: string;
  /**
   * The background color of the button. Use a color key passed in theme.colors.
   */
  color?: string;
  /**
   * The variant of the button style to use.
   */
  variant?: 'outline' | 'ghost' | 'link' | 'solid' | 'none';
  /**
   * If true, the button will be disabled.
   */
  isDisabled?: boolean;
  /**
   * If true, the button will show a spinner.
   */
  isLoading?: boolean;
  /**
   * The label to show in the button when isLoading is true. If no text is passed, it only shows the spinner
   */
  loadingText?: string;
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'none';
  /**
   * Other tailwind utility classes or custom classnames you wish to include
   */
  className?: string;
  /**
   * The lightness or how dark you want the background color to be
   */
  children?: React.ReactNode;
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   * defaults to "button"
   */
  component?: React.ElementType;
  id?: string;
  /**
   *  when the href prop is set it changes the tag name to a
   */
  href?: string;
  onClick?: () => any;
}

const darken = (value: string): string => {
  const splitStr = value.split('-');
  const dept = Number(splitStr[1]);
  const color = splitStr[0];

  if (dept >= 800) {
    return `${color}-${dept - 200}`;
  }

  return `${color}-${dept + 200}`;
};

const lighten = (value: string): string => {
  const color = value.split('-')[0];

  return `${color}-${100}`;
};

const Button = React.forwardRef((props: ButtonProps, ref: React.Ref<any>) => {
  const {
    component: Component = 'button',
    variant = 'solid',
    color = 'teal-500',
    text = 'white',
    size = 'md',
    children,
    ...rest
  } = props;
  const DEFAULT_CLASS = 'block focus:shadow-outline focus:outline-none rounded font-semibold flex';
  const STYLES = {
    solid: `bg-${color} text-${text} hover:bg-${darken(color)}`,
    outline: `bg-transparent hover:bg-${lighten(
      text
    )} text-${text} border border-${text} focus:border-transparent`,
    ghost: `hover:bg-${lighten(text)} text-${text}`,
    link: `hover:underline text-${text}`,
    md: 'px-4 py-2',
  };

  let BTNClass: string;

  if (variant === 'none') {
    BTNClass = '';
  } else {
    BTNClass = clx(DEFAULT_CLASS, {
      [STYLES[size]]: size && variant !== 'link',
      [STYLES[variant]]: variant,
    });
  }

  return (
    <Component className={BTNClass} ref={ref} {...rest}>
      {children ? children : 'Button'}
    </Component>
  );
});

export default Button;
