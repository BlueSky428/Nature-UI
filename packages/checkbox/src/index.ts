export * from './checkbox';
export * from './checkbox-group';
export * from './checkbox-types';
export * from './use-checkbox';
export * from './use-checkbox-group';
