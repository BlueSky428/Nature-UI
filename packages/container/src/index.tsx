import * as React from 'react';
import clx from 'clsx';

interface IContainer {
  children?: React.ReactNode;
  /**
   * Determine the max-width of the container.
   */
  size?: 'lg' | 'md' | 'sm' | 'xl' | 'xs';
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component?: React.ComponentType;
  className?: string;
}

const BASE_STYLE = 'mx-auto container';
const LG = 'px-20';
const MD = 'px-32';
const SM = 'px-48';
const XL = 'px-0';
const XS = 'px-64';

const Container: React.ForwardRefExoticComponent<IContainer> = React.forwardRef(
  (props: IContainer, ref: React.Ref<any>) => {
    const { component: Component = 'div', size = 'lg', children, className = '' } = props;

    const componentClass: string = clx(BASE_STYLE, {
      [LG]: size === 'lg',
      [MD]: size === 'md',
      [SM]: size === 'sm',
      [XL]: size === 'xl',
      [XS]: size === 'xs'
    });

    return (
      <Component className={componentClass} ref={ref}>
        <Component className={className}>{children}</Component>
      </Component>
    );
  }
);

Container.displayName = 'Container';

export default Container;
