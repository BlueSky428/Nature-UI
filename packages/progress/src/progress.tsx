/** ** */
import { generateStripe } from '@nature-ui/color';
import { clsx, css, HTMLNatureProps, nature } from '@nature-ui/system';
import { isUndefined, __DEV__ } from '@nature-ui/utils';
import React from 'react';
import {
  getProgressProps,
  GetProgressPropsOptions,
  progress,
  stripe,
} from './progress.utils';

export interface ProgressLabelProps extends HTMLNatureProps<'div'> {
  fontSize?: number;
}
/**
 * ProgressLabel (Linear)
 *
 * Progress component used to show the numeric value of the progress.
 */
export const ProgressLabel: React.FC<ProgressLabelProps> = ({
  className,
  fontSize = 12,
  ...rest
}) => {
  const _size = `${(60 / 100) * fontSize}px`;

  const DEFAULTS = css`
    top: 50%;
    left: 50%;
    width: 100%;
    text-align: center;
    position: absolute;
    transform: translate(-50%, -50%);
    font-size: ${_size};
  `;

  return <nature.div className={clsx(DEFAULTS, className)} {...rest} />;
};

if (__DEV__) {
  ProgressLabel.displayName = 'ProgressLabel';
}

export interface ProgressTrackProps
  extends HTMLNatureProps<'div'>,
    GetProgressPropsOptions {
  size?: string;
}

export type ProgressIndicatorProps = HTMLNatureProps<'div'> &
  GetProgressPropsOptions;

/**
 * ProgressTrack
 *
 * Wrapper element which houses the progress indicator and progress label.
 */
const ProgressTrack: React.FC<ProgressTrackProps> = (props) => {
  const {
    className,
    size,
    min,
    max,
    value,
    valueText,
    isIndeterminate,
    ...rest
  } = props;

  const progress = getProgressProps({
    value,
    min,
    max,
    valueText,
    isIndeterminate,
  });

  const trackStyles = clsx(className, {
    [css`
      width: ${progress.percent}%;
    `]: progress.percent,
  });
  return <nature.div className={trackStyles} {...progress.bind} {...rest} />;
};

interface ProgressOptions {
  /**
   * The `value` of the progress indicator
   * If `undefined` the progress bar will be in `indeterminate` state
   */
  value?: number;
  /**
   * The minimum value of the progress
   */
  min?: number;
  /**
   * The maximum value of the progress
   */
  max?: number;
  /**
   * If `true`, the progress bar will show stripe
   */
  hasStripe?: boolean;
  /**
   * If `true`, and hasStripe is `true`, the stripes will be animated
   */
  isAnimated?: boolean;
  /**
   * Determines the height of the progress bar
   * defaults to `md`
   */
  size?: 'sm' | 'md' | 'lg' | number;
  trackClassName?: string;
  indicatorClassName?: string;
  /**
   * If `true`, will show the percentage of progress bar
   */
  showPercent?: boolean;
  /**
   * If provided, will overwrite the percent label
   */
  valueText?: string;
}

const SIZES = {
  sm: 8,
  md: 12,
  lg: 16,
};

export interface ProgressProps
  extends ProgressOptions,
    HTMLNatureProps<'div'> {}

/**
 * Progress (Linear)
 *
 * Progress is used to display the progress status for a task that takes a long time or consists of several steps.
 *
 * It includes accessible attributes to help assistive technologies understand and speak the progress values
 */
export const Progress = (props: ProgressProps) => {
  const {
    value,
    min = 0,
    max = 100,
    hasStripe,
    isAnimated,
    children,
    size = 'md',
    indicatorClassName = 'bg-blue-500',
    valueText,
    'aria-label': ariaLabel,
    'aria-labelledby': ariaLabelledBy,
    className,
    ...rest
  } = props;

  const _size = typeof size === 'string' ? `${SIZES[size]}px` : `${size}px`;

  /**
   * Generate a stripe style for the progress bar
   */
  const stripeStyle = generateStripe();

  const isIndeterminate = isUndefined(value);
  const stripAnimation = css`
    animation: ${stripe} 1s linear infinite;
  `;

  /**
   * We should not use stripe if it's `indeterminate`
   */
  const shouldAddStripe = !isIndeterminate && hasStripe;

  const shouldAnimateStripe = shouldAddStripe && isAnimated;

  /**
   * Generate styles for stipe and stripe animation
   */
  const indeterminateStyles = css`
    position: absolute;
    will-change: left;
    min-width: 50%;
    animation: ${progress} 1s ease infinite normal none running;
  `;

  const _indicatorStyles = clsx(indicatorClassName, `h-full`, {
    [stripeStyle]: shouldAddStripe,
    [stripAnimation]: shouldAnimateStripe,
    [indeterminateStyles]: isIndeterminate,
  });

  const _height = css`
    height: ${_size};
  `;
  const trackStyles = clsx(
    'overflow-hidden relative bg-gray-300 w-full h-6',
    _height,
    className,
  );

  return (
    <nature.div className={trackStyles} {...rest}>
      <ProgressTrack
        aria-label={ariaLabel}
        aria-labelledby={ariaLabelledBy}
        value={value}
        valueText={valueText}
        min={min}
        max={max}
        className={_indicatorStyles}
        size={_size}
      />
    </nature.div>
  );
};

if (__DEV__) {
  Progress.displayName = 'Progress';
}
