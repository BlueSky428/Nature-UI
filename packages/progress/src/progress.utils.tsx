import { keyframes } from '@nature-ui/system';
import { isFunction, valueToPercent } from '@nature-ui/utils';

/**
 * CSS Animation for progress spin effect
 */
export const spin = keyframes`
0% {
stroke-dasharray: 1, 400;
stroke-dashoffset: 0;
}

50% {
    stroke-dasharray: 400, 400;
    stroke-dashoffset: -100;
}

100% {
    stroke-dasharray: 400, 400;
    stroke-dashoffset: -200;
}
`;

/**
 * CSS Animation for progress rotate effect
 */
export const rotate = keyframes`
0% {transform: rotate(0deg);}
100% {transform: rotate(360deg);}
`;

/**
 * CSS Animation for progress indeterminate effect
 */
export const progress = keyframes`
  0% { left: -40% }
  100% { left: 100% }
`;

/**
 * CSS Animation for progress stripe effect
 */
export const stripe = keyframes`
from {background-position: 1rem 0;}
too {background-position: 0 0;}
`;

export interface GetProgressPropsOptions {
  value?: number;
  min: number;
  max: number;
  valueText?: string;
  getValueText?(value?: number, percent?: number): string;
  isIndeterminate?: boolean;
}

/**
 * Get the common `aria-*` attributes for both the linear and circular
 * progress components
 */
export const getProgressProps = (options: GetProgressPropsOptions) => {
  const {
    value = 0,
    min,
    max,
    valueText,
    getValueText,
    isIndeterminate,
  } = options;

  const percent = valueToPercent(value, min, max);

  const getAriaValueText = () => {
    if (value == null) return undefined;
    return isFunction(getValueText) ? getValueText(value, percent) : valueText;
  };

  const bind = {
    'data-indeterminate': isIndeterminate ? '' : undefined,
    'aria-valuemax': max,
    'aria-valuemin': min,
    'aria-valuenow': isIndeterminate ? undefined : value,
    'aria-valuetext': getAriaValueText(),
    role: 'progressbar',
  };

  return {
    bind,
    percent,
    value,
  };
};
