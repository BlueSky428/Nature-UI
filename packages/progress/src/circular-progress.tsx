/** ** */
import { clsx, css, HTMLNatureProps, nature } from '@nature-ui/system';
import { isUndefined, StringOrNumber, __DEV__ } from '@nature-ui/utils';
import React from 'react';
import { getProgressProps, rotate, spin } from './progress.utils';

type CircleProps = HTMLNatureProps<'circle'> & {
  /**
   * The color name of the progress track.
   */
  trackColor?: string;
  /**
   * The color of the progress indicator.
   */
  colorScheme?: string;
  _transition?: string;
  _animation?: string;
};

/**
 * Circle
 *
 * SVG circle element visually indicating the shape of the component
 */
const Circle = (props: CircleProps) => {
  const {
    className = '',
    trackColor,
    colorScheme,
    _animation,
    _transition,
    ...rest
  } = props;

  const _className = clsx(
    `stroke-current text-${trackColor || colorScheme}`,
    {
      [className]: className,
    },
    _animation,
    _transition,
  );

  return (
    <nature.circle
      className={_className}
      cx={50}
      cy={50}
      fill='transparent'
      r={42}
      {...rest}
    />
  );
};

if (__DEV__) {
  Circle.displayName = 'Circle';
}

type ShapeProps = HTMLNatureProps<'svg'> & {
  size?: StringOrNumber;
  isIndeterminate?: boolean;
  children?: React.ReactNode;
};

/**
 * Shape
 *
 * SVG wrapper element for the component's circular shape
 */
const Shape = (props: ShapeProps) => {
  const { size, isIndeterminate, className, ...rest } = props;

  const _className = clsx(
    {
      [css`
        animation: ${rotate} 2s linear infinite;
      `]: isIndeterminate,
    },
    className,
  );

  return (
    <nature.svg
      className={_className}
      width={size}
      height={size}
      viewBox='0 0 100 100'
      {...rest}
    />
  );
};

if (__DEV__) {
  Shape.displayName = 'Shape';
}

interface CircularProgressOptions {
  /**
   * The size of the circular progress in CSS units
   */
  size?: StringOrNumber;
  /**
   * Maximum value defining 100% progress made (must be higher than 'min')
   */
  max?: number;
  /**
   * Minimum value defining 'no progress' (must be lower than 'max')
   */
  min?: number;
  /**
   * The thickness of progress indicator as a ratio of `size`. Must be between `0` and `1`
   */
  thickness?: StringOrNumber;
  /**
   * Current progress (must be between min/max)
   */
  value?: number;
  /**
   * If `true`, the cap of the progress indicator will be rounded.
   */
  capIsRound?: boolean;
  /**
   * The content of the circular progress bar. If passed, the content will be inside and centered in the progress bar.
   */
  children?: React.ReactNode;
  /**
   * The color name of the progress track.
   */
  trackColor?: string;
  /**
   * The color of the progress indicator.
   */
  colorScheme?: string;
  /**
   * The desired valueText to use in place of the value and also displayed within the circular-progress
   */
  valueText?: string;
  /**
   * If `true`, will show the percentage of progress bar
   */
  showPercent?: boolean;
  /**
   * A function that returns the desired valueText to use in place of the value
   */
  getValueText?(value?: number, percent?: number): string;
  /**
   * If `true`, the progress will be indeterminate and the `value`
   * prop will be ignored
   */
  isIndeterminate?: boolean;
}

export type CircularProgressProps = HTMLNatureProps<'div'> &
  CircularProgressOptions;

/**
 * CircularProgressLabel
 *
 * CircularProgress component label. In most cases it's a numeric indicator
 * of the circular progress component's value
 */
export const CircularProgressLabel = (
  props: HTMLNatureProps<'div'> & {
    fontSize?: StringOrNumber;
  },
) => {
  const { className = '', fontSize = '0.7rem', ...rest } = props;

  const STYLES = css`
    top: 50%;
    left: 50%;
    width: 100%;
    text-align: center;
    position: absolute;
    transform: translate(-50%, -50%);
    font-size: ${fontSize};
  `;

  const _className = clsx(STYLES, {
    [className]: className,
  });

  return <nature.div className={_className} {...rest} />;
};

/**
 * React component used to indicate the progressof an activity
 *
 * It's built using `svg` and `circle` components.
 */
export const CircularProgress: React.FC<CircularProgressProps> = (props) => {
  const {
    size = '32px',
    max = 100,
    min = 0,
    valueText,
    getValueText,
    value,
    capIsRound,
    children,
    thickness = '10px',
    colorScheme = 'blue-500',
    trackColor = 'gray-300',
    showPercent,
    isIndeterminate,
    className,
    ...rest
  } = props;

  const progress = getProgressProps({
    min,
    max,
    value,
    valueText,
    getValueText,
    isIndeterminate,
  });

  const _size = typeof size === 'string' ? size : `${size}px`;

  const determinant = isIndeterminate
    ? undefined
    : (progress.percent ?? 0) * 2.64;

  const strokeDasharray = isUndefined(determinant)
    ? undefined
    : `${determinant} ${264 - determinant}`;

  const _animation = css`
    animation: ${spin} 1.5s linear infinite;
  `;

  const _transition = css`
    transition: stroke-dasharray 0.6s ease 0s, stroke 0.6s ease;
  `;

  const indicatorProps = isIndeterminate
    ? {
        _animation,
      }
    : {
        strokeDashoffset: 66,
        strokeDasharray,
        _transition,
      };

  const _fontSize = `${(20 / 100) * parseFloat(_size)}px`;

  return (
    <nature.div
      className={clsx('inline-block relative align-middle', className)}
      {...progress.bind}
      {...rest}
    >
      <Shape size={_size} isIndeterminate={isIndeterminate}>
        <Circle trackColor={trackColor} strokeWidth={thickness} />

        <Circle
          colorScheme={colorScheme}
          strokeWidth={thickness}
          strokeLinecap={capIsRound ? 'round' : undefined}
          {...indicatorProps}
        />
      </Shape>
      {showPercent && !valueText ? (
        <CircularProgressLabel fontSize={_fontSize}>
          {value}%
        </CircularProgressLabel>
      ) : valueText ? (
        <CircularProgressLabel fontSize={_fontSize}>
          {valueText}
        </CircularProgressLabel>
      ) : (
        children
      )}
    </nature.div>
  );
};

if (__DEV__) {
  CircularProgress.displayName = 'CircularProgress';
}
