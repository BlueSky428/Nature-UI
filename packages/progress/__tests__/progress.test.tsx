import { render } from '@nature-ui/test-utils';
import { CircularProgress, Progress } from '../src';

describe('@nature-ui/circular-progress', () => {
  test('CircularProgress renders correctly', () => {
    const { asFragment } = render(
      <div>
        <CircularProgress size='60px' value={20} />
        <CircularProgress
          size='120px'
          trackColor='transparent'
          thickness={10}
          value={60}
        />
      </div>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('CircularProgress: has the proper aria, data, and role attributes', () => {
    const props = {
      trackColor: 'transparent',
      label: 'value',
      thickness: 10,
      value: 20,
    };
    const utils = render(<CircularProgress {...props} />);
    const progress = utils.getByRole('progressbar');

    expect(progress).not.toHaveAttribute('data-indeterminate');
    expect(progress).toHaveAttribute('aria-valuemax', '100');
    expect(progress).toHaveAttribute('aria-valuemin', '0');
    expect(progress).toHaveAttribute('aria-valuenow', '20');
    expect(progress).toHaveAttribute('aria-valuetext', 'value');

    // rerender as indeterminate
    utils.rerender(<CircularProgress {...props} value={undefined} />);

    expect(progress).toHaveAttribute('data-indeterminate');
    expect(progress).not.toHaveAttribute('aria-valuenow');

    // rerender with getValueText function
    utils.rerender(
      <CircularProgress
        {...props}
        getValueText={(value: any, percent: any) => `${value} (${percent}%)`}
      />,
    );

    expect(progress).toHaveAttribute('aria-valuetext', '20 (20%)');
  });

  test('Progress renders correctly', () => {
    const { asFragment } = render(
      <div>
        <Progress color='green' size='sm' value={20} />
        <Progress color='blue' size='md' hasStripe value={40} />
        <Progress color='yellow' size='lg' hasStripe isAnimated value={80} />
      </div>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('Progress: has the proper aria, data, and role attributes', () => {
    const utils = render(<Progress color='green' size='sm' value={20} />);
    const progress = utils.getByRole('progressbar');

    expect(progress).not.toHaveAttribute('data-indeterminate');
    expect(progress).toHaveAttribute('aria-valuemax', '100');
    expect(progress).toHaveAttribute('aria-valuemin', '0');
    expect(progress).toHaveAttribute('aria-valuenow', '20');
    expect(progress).not.toHaveAttribute('aria-valuetext');

    // rerender as indeterminate
    utils.rerender(<Progress color='green' size='sm' value={undefined} />);

    expect(progress).toHaveAttribute('data-indeterminate');
    expect(progress).not.toHaveAttribute('aria-valuenow');
  });
});
