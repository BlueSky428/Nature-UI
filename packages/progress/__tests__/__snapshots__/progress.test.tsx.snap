// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nature-ui/circular-progress CircularProgress renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="20"
      class="inline-block relative align-middle"
      role="progressbar"
    >
      <svg
        class=""
        height="60px"
        viewBox="0 0 100 100"
        width="60px"
      >
        <circle
          class="stroke-current text-gray-200"
          cx="50"
          cy="50"
          fill="transparent"
          r="42"
          stroke-width="10px"
        />
        <circle
          class="stroke-current text-blue-400 css-e0aiw1"
          cx="50"
          cy="50"
          fill="transparent"
          r="42"
          stroke-dasharray="52.800000000000004 211.2"
          stroke-dashoffset="66"
          stroke-width="10px"
        />
      </svg>
    </div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="60"
      class="inline-block relative align-middle"
      role="progressbar"
    >
      <svg
        class=""
        height="120px"
        viewBox="0 0 100 100"
        width="120px"
      >
        <circle
          class="stroke-current text-transparent"
          cx="50"
          cy="50"
          fill="transparent"
          r="42"
          stroke-width="10"
        />
        <circle
          class="stroke-current text-blue-400 css-e0aiw1"
          cx="50"
          cy="50"
          fill="transparent"
          r="42"
          stroke-dasharray="158.4 105.6"
          stroke-dashoffset="66"
          stroke-width="10"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`@nature-ui/circular-progress Progress renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="overflow-hidden relative bg-gray-300 w-full h-6 css-jb16r4"
      color="green"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="20"
        class="h-full bg-teal-400 css-nyci5l"
        color="green"
        role="progressbar"
      />
    </div>
    <div
      class="overflow-hidden relative bg-gray-300 w-full h-6 css-1r10t72"
      color="blue"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="40"
        class="h-full bg-teal-400 css-1jz7m94 css-1ttdosy"
        color="blue"
        role="progressbar"
      />
    </div>
    <div
      class="overflow-hidden relative bg-gray-300 w-full h-6 css-14dp4mi"
      color="yellow"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="80"
        class="h-full bg-teal-400 css-1jz7m94 css-1g5oqvd css-13r1wu6"
        color="yellow"
        role="progressbar"
      />
    </div>
  </div>
</DocumentFragment>
`;
